File: CodeWars/reverse_words.js
function reverseWords(str) {
    return str
        .split(" ") // separate using space as separator
        .map( // for every item
            word =>
                word // return the word
                    .split("") // characters of the word in an array
                    .reverse() // reverse the array
                    .join("") // join all the items in the array with no separator
        )
        .join(" ") // join all the words with spaces between them
}

File: CodeWars/reverse_words.py
"""
https://www.codewars.com/kata/5259b20d6021e9e14c0010d4/train/python

Complete the function that accepts a string parameter, and reverses each word in the string. All spaces in the string should be retained.
Examples

"This is an example!" ==> "sihT si na !elpmaxe"
"double  spaces"      ==> "elbuod  secaps"

"""

def reverse_words(text):
    words = text.split(" ")
    reversed_words = [word[::-1] for word in words]
    
    return " ".join(reversed_words)
